### PR TITLE
Bug 1881905: Adds gatherer for PodDistributionBudget

### PR DESCRIPTION
**Pull request by 0sewa0**

### CODE REVIEW

File: pkg/gather/clusterconfig/clusterconfig.go [hunk 1]
```diff
@@ -1091,6 +1117,20 @@ func anonymizePod(pod *corev1.Pod) *corev1.Pod {
 	return pod
 }
 
+type PodDisruptionBudgetsAnonymizer struct {
+	*policyv1beta1.PodDisruptionBudget
+}
+
+// Marshal implements serialization of a Pod with anonymization
```
Comment (tremes) [inline pkg/gather/clusterconfig/clusterconfig.go:1124]: Copy/paste comment. It's just minor but is there anything we could possibly anonymize (some IDs, IP addressess and similar)?

Comment (0sewa0) [reply pkg/gather/clusterconfig/clusterconfig.go:1124]: Yeah the comments are a mess but its WIP for a reason :D

About what to anonymize:  No idea really, the ones that I saw had nothing in them that I would consider confidential. maybe the `metadata.managedFields` can have something confidential info.

**Comment by 0sewa0** [comment]
currently some tests fail because:
`Sep 23 11:17:12.821: FAIL: Some cluster operators are not ready: insights (Degraded=True PeriodicGatherFailed: Source config could not be retrieved: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:openshift-insights:operator" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope)`

@iNecas @martinkunc @natiiix any thoughts ?

**Comment by martinkunc** [comment]
Did this passed some local tests ? Basically you should get a cluster (quicklab) run IO and check data in /tmp folder.
Anyway, I think you need to add the Role policy according to this comment: https://github.com/rancher/rancher/issues/20325#issuecomment-496798677
in this file:
https://github.com/openshift/insights-operator/blob/master/manifests/03-clusterrole.yaml#L64

When you change file locally, you can try
```
oc apply -f 03-clusterrole.yaml
```
To apply the change to see if it works.

File: pkg/gather/clusterconfig/clusterconfig.go (hunk 1 above)
Comment (martinkunc) [reply pkg/gather/clusterconfig/clusterconfig.go]: Here is the type definition for PDB. To me it doesn't look like there could be anything worth anonymizing. What do you think ?
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/policy/types.go

Comment (0sewa0) [reply pkg/gather/clusterconfig/clusterconfig.go]: that was my impression as well

**Review by natiiix — COMMENTED** [review]
There are quite a few commits. It's usual to squash them to reduce the number of commits in the upstream branches. 1-3 is usually okay, but 6 seems like too many for a single enhancement.

**Comment by 0sewa0** [comment]
@natiiix Can't github do it ?? Gitlab made it all easy :(

**Comment by natiiix** [comment]
> @natiiix Can't github do it ?? Gitlab made it all easy :(

@0sewa0 GitHub can do it, but the merge bot does as the name suggests, creates a merge commit, not a squash. Sometimes it's absolutely necessary for cherry-picking because you don't want to backport documentation or configuration changes in some scenarios.

**Comment by martinkunc** [comment]
Nice, thank you @0sewa0 !
/lgtm

**Pull request merged**